### ISSUE TITLE
May I get the train code?

### ISSUE THREAD

**Issue by YeonjeeJung** [soyoung97/ListT5#2]
Hi, I'm very interested in your work, and I would like to know about the loss function you used.
I cannot find the loss function in the paper. Did you use the same loss function as RankT5?

And if you don't mind, I want to use this method in my work. So I would be very grateful if you could upload the train code you used.
Thanks a lot!

**Comment by soyoung97** [comment]
Hi @YeonjeeJung , thank you for your interest in my work!

I simply used the CrossEntropyLoss in torch to train ListT5, where the label is the gold ordering represented in text format (e.g., "2 1 3 4 5").
I did also experiment with other type of loss, such as Poly1 Loss or listwise ranking loss. (code below)
However, the model had enough performance gains compared with baseline models (RankT5, MonoT5, ..), so I didn't investigate with the loss function any further and just used the simple torch.nn.CrossEntropyLoss. I think the performance may improve by introducing more advanced loss function!


I'm sorry for the delay but I didn't manage to clean up the training code yet.. but here is the screenshot of the main data processing and training parts of the code! You can get the training data (input / output gold label data format) by requesting on the hf datasets page :)
https://huggingface.co/datasets/Soyoung97/ListT5-train-data
# Input data processing code
<img width="1013" alt="image" src="https://github.com/user-attachments/assets/8b5852da-1f98-4aba-8b72-164656aff379">

# Training code 
<img width="860" alt="image" src="https://github.com/user-attachments/assets/5a30b000-e213-4872-8e76-b0339d342f7b">

**Comment by YeonjeeJung** [comment]
It would be very helpful for me. I have learnt a lot from it!

Thank you so much for your rapid reply and providings!
